### ISSUE TITLE
🐛 VS Code/Cursor IDE shows useHookAtTopLevel error despite rule being disabled in config overrides

### ISSUE THREAD

**Issue by waseemshahwan** [biomejs/biome-vscode#766]
### VS Code version

Cursor IDE: 1.5.5, VSCode: 1.99.3

### Extension version

2025.7.41733

### Biome version

2.2.2

### Operating system

- [ ] Windows
- [x] macOS
- [ ] Linux

### Description

In a monorepo setup, when working on an individual package that has its own biome.json configuration extending the root config, the Biome extension in the IDE incorrectly reports a `useHookAtTopLevel` error. This occurs even though the rule is explicitly disabled for `*.ts` files in the package's configuration overrides. The issue only affects the IDE extension when the workspace is opened at the individual package level within the monorepo. The CLI correctly respects the configuration override and does not report this error.

Additional Context:
- The backend project has React as a dependency, which triggers the React-specific rules
- The workspace is opened at the package level (`/packages/tool/dev`), not at the monorepo root
- The `.vscode/settings.json` has been configured to point to the correct Biome binary path and `biome.json` configuration path (verified via output logs)
- The monorepo uses Bun workspaces: https://bun.com/docs/install/workspaces

### Steps to reproduce

1. Create a Bun monorepo with Biome installed at the root
2. Create a package with its own biome.json that extends the root config
3. Add an override to disable `useHookAtTopLevel` for `*.ts` files:
```json
{
  "$schema": "https://biomejs.dev/schemas/2.2.2/schema.json",
  "extends": "//",
  "root": false,
  "overrides": [
    {
      "includes": ["**/*.ts"],
      "linter": {
        "rules": {
          "correctness": {
            "useHookAtTopLevel": "off"
          }
        }
      }
    }
  ]
}
```
4. Create a TypeScript file with a function starting with "use" that has conditional logic:
```ts
const useContainersLockfile = async (timeout: number): Promise<void> => {
  // ... some code
  if (timeout <= 0) {
    throw new Error("Timeout waiting for containers lockfile");
  }
  await new Promise((resolve) => setTimeout(resolve, 10));
  return useContainersLockfile(timeout - 10);
};
```
5. Open the file in VS Code/Cursor with the Biome extension

### Expected behavior

The IDE should not show the `useHookAtTopLevel` error since it's disabled for TypeScript files in the configuration overrides, matching the CLI behavior.

### Does this issue occur when using the CLI directly?

No

### Logs

```plain text

```

**Comment by ematipico** [comment]
> The .vscode/settings.json has been configured to point to the correct Biome binary path and biome.json configuration path (verified via output logs)

We really need these settings as part of the reproduction.
